### ISSUE TITLE
Could not subclass Codeception\Extension\MultiDb - Fatal error: Call to private method normaliseAsIs()

### ISSUE THREAD

**Issue by aurum86** [redmatter/Codeception-MultiDb#20]
Hello,

MultiDb works fine if I use it directly. But if I want to extend a subclass from it, I get the following:

> Fatal error: Call to private method Codeception\Extension\MultiDb::normaliseAsIs() from context 'Codeception\Extension\DbHelper' in /vendor/redmatter/codeception-multidb/src/Codeception/Extension/MultiDb.php on line 966

By **changing visibility** of private static function normaliseAsIs **from private to protected** **solves the issue**. Any chance of this hot fix to be included before final release of 2.0.0 ?

Reproduced using "redmatter/codeception-multidb": "dev-master"

**Comment by codemedic** [comment]
Hi @aurum86 

If the change is to fix something that needs fixing, I will be happy to accommodate that.
Before that, please can you explain what change of behaviour are you looking to implement in  your own version of `normaliseAsIs()`? It might benefit others too, if that can be ported into the base implementation.

Thanks

**Comment by aurum86** [comment]
The thing is, I do not want to change function "normaliseAsIs" behavior. I would like to add more methods into MutliDb module which are specific to my own Application and thus could not be applied to public audience. To add those additional methods I have to extend MultiDb class. But then I could not use my MultiDbExtended because it refers to that private static method... By simply changing the visibility, it would solve the problem of mine and perhaps others who want to subclass MultiDb by any reason.

You could just simply Extend a blank class from MultiDb and try to use it in Codeception tests and should get the same error.

**Comment by abulford** [comment]
@aurum86: are you trying to call `normaliseAsIs` directly from your child class?  I don't think making the `normaliseAsIs` function protected is necessarily the right answer here, possibly a better solution would be to move the function to the `AsIs` class and make it public there.

Please feel free to submit a pull request including this change.

**Comment by aurum86** [comment]
@abulford No, no direct calls. As I said earlier, it is the same error even if I use totally empty extended class. Example for a proof:

DbHelper.php:
```
namespace Functional;
class DbHelper extends MultiDb {
  // no code here
}
```

Functional.suite.yml:
```
class_name: FunctionalTester
modules:
  enabled:
    - Functional\DbHelper
  config:
    Functional\DbHelper:
      connectors:
        ....what ever...
```

Please, disregard my custom namespace Functional\DbHelper.

This is basically enough to get this error and also shows that its impossible to extend any custom functionality this way because of that error. But I would agree with you to move the method to AsIs class and make it public.
Since I could not wait for this to be fixed and really needed this to be working (MultiDb extension is cool in overall !), I solved it in a hacky way - redeclared the same private static method in my DbHelper class.
